src/main/java/io/r2dbc/postgresql/codec/Point.java

```diff
@@ -0,0 +1,102 @@
+package io.r2dbc.postgresql.codec;
+
+/**
+ * <p>It maps to the point datatype in org.postgresql.</p>
+ *
+ * <p>This implements a version of java.awt.Point, except it uses double to represent the coordinates.</p>
+ */
+public class Point {
+    public double x;
```

Comment:
Do we really need this to be mutable?

Reply:
I referred [PGpoint](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/geometric/PGpoint.java) of pgjdbc. there it is mentioned as public. I think that perhaps it represents a 2D-point, hence, instead of adding both getters & setters, this value is made mutable.

Reply:
I can make it private and generate a getter/setter for it.

Reply:
Those should be rather immutable fields (`private final` without setters). Thinking of values as atomic and immutable value types is useful as a single point (and future geo types) cannot be changed partially but each update to a point requires a new value object instance.